server/parser.go

```diff
@@ -160,7 +159,24 @@ func (c *client) parse(buf []byte) error {
 			c.op = b
 			if b != 'C' && b != 'c' {
 				if authSet {
-					goto authErr
+					if s == nil {
+						goto authErr
+					}
+					var ok bool
+					// Check here for NoAuthUser. If this is set allow non CONNECT protos as our first.
+					// E.g. telnet proto demos.
+					if noAuthUser := s.getOpts().NoAuthUser; noAuthUser != _EMPTY_ {
+						s.mu.Lock()
+						user, exists := s.users[noAuthUser]
+						s.mu.Unlock()
+						if exists {
+							c.RegisterUser(user)
+							authSet, ok = false, true
```

Comment:
But again, this seem wrong. As long as there is no_auth_user, even if a client plans to send CONNECT, then it will always be registered as a no_auth_user...

Reply:
Nevermind, if a client sends CONNECT, we will not be there of course...